src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,45 +1,96 @@
-import React from 'react';
-import { Loader } from '../Loader';
+import React, { useState, useEffect, SetStateAction } from 'react';
+// import { getUsers } from '../../services/getUsers';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import { Loader } from '../Loader/Loader';
+
+interface Props {
+  todoCard: Todo | undefined;
+  setIsModalOpen: React.Dispatch<SetStateAction<boolean>>
+  setIconId: React.Dispatch<SetStateAction<number>>
+}
+
+export const TodoModal: React.FC<Props> = ({
+  todoCard,
+  setIsModalOpen,
+  setIconId,
+}) => {
+  const [selectedUser, setsSelectedUser] = useState<User>();
+  const [isUserLoad, setIsUserLoad] = useState(false);
+
+  useEffect(() => {
+    setIsUserLoad(true);
+    setTimeout(() => {
+      return fetch('https://mate-academy.github.io/'
+        + `react_dynamic-list-of-todos/api/users/${todoCard?.userId}.json`)
+        .then((response) => {
+          return response.json();
+        }).then((user) => {
+          setsSelectedUser(user);
+        }).finally(() => setIsUserLoad(false));
+    }, 300);
+  }, [todoCard]);
+
+  const handleCloseCard = () => {
+    setIsModalOpen(false);
+    setIconId(0);
+  };
 
-export const TodoModal: React.FC = () => {
   return (
+
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
+      {isUserLoad ? (
+        <>
+          <Loader />
+        </>
 
-      {true ? (
-        <Loader />
       ) : (
         <div className="modal-card">
           <header className="modal-card-head">
             <div
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              Todo #
+              {todoCard?.id}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
               type="button"
               className="delete"
               data-cy="modal-close"
+              onClick={handleCloseCard}
             />
           </header>
 
           <div className="modal-card-body">
             <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
+              {todoCard?.title}
             </p>
 
             <p className="block" data-cy="modal-user">
               {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
+              {todoCard?.completed ? (
+                <strong
+                  className="has-text-success"
+                >
+                  Done
+                </strong>
```

Comment:
```suggestion
                <strong className="has-text-success">Done</strong>
```

this code is quite short, so we can put it in one line